### PR TITLE
Update to newer version of the FTL.

### PR DESCRIPTION
**Pull request by pylaligand**

### CODE REVIEW

File: shell/common/switches.cc [hunk 1]
```diff
@@ -78,13 +80,13 @@ void PrintUsage(const std::string& executable_name) {
   std::cerr << std::string(column_width, '-') << std::endl;
 }
 
-const char* FlagForSwitch(Switch swtch) {
+const ftl::StringView FlagForSwitch(Switch swtch) {
```
Comment (pylaligand) [inline shell/common/switches.cc]: Looking at this PR it seems I could keep the return type here and simply return `gSwitchDescs[i].flag.data()` on line 86, and pretty much the rest of the diff would go away. WDYT?

Comment (chinmaygarde) [inline shell/common/switches.cc]: What is the point of this patch?

Comment (pylaligand) [inline shell/common/switches.cc]: Yes, that was my suggestion. The FTL update is the point, it allows all this code to build successfully on Fuchsia.

Comment (pylaligand) [inline shell/common/switches.cc]: Actually the problem is that the result of `FlagForSwitch` is used as an input to several FTL APIs (e.g. `command_line.GetOptionValue`), and these APIs require a `StringView`. If you want to keep `char*`, I can create `StringView` objects only for the purpose of calling the FTL APIs.

Comment (pylaligand) [inline shell/common/switches.cc]: Which approach do you prefer?

Comment (chinmaygarde) [inline shell/common/switches.cc]: Either approach is fine by me.

**Comment by cbracken** [comment]
Looks like may have resulted in a Windows engine failure:
https://build.chromium.org/p/client.flutter/builders/Windows%20Engine/builds/381
```
[----------] 2 tests from File
[ RUN      ] File.GetFileSize
[       OK ] File.GetFileSize (29 ms)
[ RUN      ] File.WriteFileInTwoPhases
e:\b\build\slave\windows_engine\build\src\lib\ftl\files\file_unittest.cc(36): error: Value of: WriteFileInTwoPhases(path, content, dir.path())
  Actual: false
Expected: true
[  FAILED  ] File.WriteFileInTwoPhases (17 ms)
[----------] 2 tests from File (46 ms total)
```

**Comment by cbracken** [comment]
Looks like Mac engine broke too:
https://build.chromium.org/p/client.flutter/builders/Mac%20Engine/builds/1462

**Comment by tvolkert** [comment]
Can we please revert this to get the engine build green again, then roll forward with a fix?

**Comment by cbracken** [comment]
https://github.com/flutter/engine/pull/3792 reverts

**Pull request merged**